superset/utils/dashboard_filter_scopes_converter.py

```diff
@@ -88,3 +90,249 @@ def copy_filter_scopes(
+                                    "multiSelect": config.get(
+                                        "multiple",
+                                        False,
+                                    ),
+                                    "searchAllOptions": config.get(
+                                        "searchAllOptions",
+                                        False,
+                                    ),
+                                },
+                                "filterType": "filter_select",
+                                "name": config.get("label") or field,
+                                "targets": [
+                                    {
+                                        "column": {"name": field},
+                                        "datasetId": filter_box.datasource_id,
+                                    },
+                                ],
+                            }
+                        )
+
+                        if "metric" in config:
+                            fltr["sortMetric"] = config["metric"]
+                            fltr["controlValues"]["sortAscending"] = config["asc"]
+
+                        if params.get("adhoc_filters"):
+                            fltr["adhoc_filters"] = params["adhoc_filters"]
+
+                        # Pre-filter available values based on time range/column.
+                        time_range = params.get("time_range")
+
+                        if time_range and time_range != "No filter":
+                            fltr.update(
+                                {
+                                    "time_range": time_range,
+                                    "granularity_sqla": params.get("granularity_sqla"),
+                                }
+                            )
+
+                        if not default:
+                            default = config.get("defaultValue")
+
+                            if default:
+                                if config["multiple"]:
+                                    default = default.split(";")
+                                else:
+                                    default = [default]
+
+                        if default:
+                            fltr["defaultDataMask"] = {
+                                "extraFormData": {
+                                    "filters": [
+                                        {
+                                            "col": field,
+                                            "op": "IN",
+                                            "val": default,
+                                        }
+                                    ],
+                                },
+                                "filterState": {"value": default},
+                            }
+
+                        break
+
+            if "filterType" in fltr:
+                filter_by_key_and_field[key][field] = fltr
+
+    # Ancestors of filter-box charts.
```

Comment:
Maybe you could extract some sub-functions to improve readability? I was thinking one function for assembling the filters and other for establishing the hierarchy. WDYT?

Reply:
@michael-s-molina I generally agree that long functions aren't great and though I took a pass at extracting this into sub functions the refactor wasn't great, i.e., assembling the filters really contains the hierarchical wiring.

I also previous refactored fetching the defaults into a function, but there needed to be lots of special case handling due to where/when the defaults were used, i.e., a `filter_select` filter may require the time range/granularity but these defaults are at the chart rather than dashboard level.

The TL;DR is granted the formulation isn't great I sense it's acceptable, especially in the context of a migration.